packages/cli/test/integration/processed-data-managers/processed-data-helper.test.ts

```diff
@@ -0,0 +1,486 @@
+import { ProcessedDataManager } from 'n8n-core';
+import type { ICheckProcessedContextData, INodeTypeData } from 'n8n-workflow';
+import type { ICheckProcessedOutput, INode, ProcessedDataItemTypes } from 'n8n-workflow';
+import { Workflow } from 'n8n-workflow';
+
+import { LoadNodesAndCredentials } from '@/load-nodes-and-credentials';
+import { NodeTypes } from '@/node-types';
+import { mockInstance } from '@test/mocking';
+import { createWorkflow } from '@test-integration/db/workflows';
+
+import { getProcessedDataManager } from '../../../src/processed-data-managers';
+import * as testDb from '../shared/test-db';
+
+let workflow: Workflow;
+
+jest.mock('../../../src/telemetry');
+
+const MOCK_NODE_TYPES_DATA = mockNodeTypesData(['set']);
+mockInstance(LoadNodesAndCredentials, {
+	loaded: {
+		nodes: MOCK_NODE_TYPES_DATA,
+		credentials: {},
+	},
+});
+function mockNodeTypesData(
+	nodeNames: string[],
+	options?: {
+		addTrigger?: boolean;
+	},
+) {
+	return nodeNames.reduce<INodeTypeData>((acc, nodeName) => {
+		return (
+			(acc[`n8n-nodes-base.${nodeName}`] = {
+				sourcePath: '',
+				type: {
+					description: {
+						displayName: nodeName,
+						name: nodeName,
+						group: [],
+						description: '',
+						version: 1,
+						defaults: {},
+						inputs: [],
+						outputs: [],
+						properties: [],
+					},
+					trigger: options?.addTrigger ? async () => undefined : undefined,
+				},
+			}),
+			acc
+		);
+	}, {});
+}
+const node: INode = {
+	id: 'uuid-1234',
+	parameters: {},
+	name: 'test',
+	type: 'test.set',
+	typeVersion: 1,
+	position: [0, 0],
+};
+
+beforeAll(async () => {
+	await testDb.init();
+
+	const node: INode = {
+		id: 'uuid-1234',
+		parameters: {},
+		name: 'test',
+		type: 'test.set',
+		typeVersion: 1,
+		position: [0, 0],
+	};
+
+	const nodeTypes = mockInstance(NodeTypes);
+	const workflowEntityOriginal = await createWorkflow();
+
+	workflow = new Workflow({
+		id: workflowEntityOriginal.id,
+		nodes: [node],
+		connections: {},
+		active: false,
+		nodeTypes,
+	});
+
+	const processedDataManager = await getProcessedDataManager();
+	await ProcessedDataManager.init(processedDataManager);
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['ProcessedData']);
+});
+
+afterAll(async () => {
+	await testDb.terminate();
+});
+
+describe('ProcessedDataManagers.ProcessedDataHelper', () => {
```

Comment:
Nice, very comprehensive tests 👌

Reply:
thanks to @janober I copied most tests over from his original PR